// pages/create-article/index.js
import React, { useEffect, useState } from 'react'
import {Redirect} from 'react-router-dom'
import ArticleForm from 'components/articleForm'
import useFetch from 'hooks/useFetch'
import {CurrentUserContext} from 'contexts/currentUser'
import { useContext } from 'react'


const CreateAticle = () => {
    const apiUrl = '/articles'
    const [{error, response}, doFetch]= useFetch(apiUrl)
    const [currentUserState] = useContext(CurrentUserContext)

    const initialValues = {
        title: '',
        description: '',
        body: '',
        tagList: []
    }

    const [isSuccessfullSubmit, setIsSuccessfullSubmit] = useState(false)
    const handleSubmit = article => {

        doFetch({
            method: 'post',
            data: {
                article
            }
        })
    }

    useEffect(()=>{
        if(!response){
            return
        }
        setIsSuccessfullSubmit(true)
    }, [setIsSuccessfullSubmit, response])

    if(currentUserState.isLoggedIn === false) {
        return <Redirect to="/" />
    }

    if(isSuccessfullSubmit){
        return <Redirect to={`/articles/${response.article.slug}`} />
    }
    return (
        <div>
            <ArticleForm errors={(error && error.errors) || {}} initialValues={initialValues} onSubmit={handleSubmit} />
        </div>
    )
}

export default CreateAticle
